### PR TITLE
Feature/57474 persist qa code route use in payload

### PR DESCRIPTION
**Pull request by seastate6**
# QR Code usage data collection

* QrCodeArrival and QrCodeSubsequentAppUse events are now sent in the payload and are stored in the mtc_results.event able against the check.
* New event LoginSuccess on successful sign-in to the pupil app.  This makes it easier to read the QR code data as well as providing useful auditing.

## Supporting changes

* MonotonicTime can now be set to a known date / sequence number
* MonotonicTime includes a static comparator function for sorting purposes

### CODE REVIEW

File: pupil-spa/src/app/login/login.component.ts [hunk 1]
```diff
@@ -101,6 +103,7 @@ export class LoginComponent implements OnInit, AfterViewInit {
             // only set the cookie for live checks
             this.deviceService.setupDeviceCookie()
           }
+          this.qrCodeUsageService.postLoginHook()
```
Comment (jon-shipley) [inline pupil-spa/src/app/login/login.component.ts:106]: This is doing much of the work.

**Review by GuyHarwood — COMMENTED** [review]
LGSF

File: pupil-spa/src/app/question/question.component.spec.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ describe('QuestionComponent', () => {
   let storageServiceSetAnswerSpy: jasmine.Spy
 
   beforeEach(waitForAsync(() => {
-    storageService = new StorageServiceMock() as StorageService
+    storageService = new StorageServiceMock() as unknown as StorageService
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply pupil-spa/src/app/question/question.component.spec.ts:35]: Yes - to 'force' StorageServiceMock to be a StorageService.

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**